android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt

```diff
@@ -21,6 +25,94 @@ open class LogtoClient(
 
     fun isAuthenticated() = idToken != null
 
+    fun getAccessToken(callback: RetrieveCallback<AccessToken>) =
+        getAccessToken(null, null, callback)
+
+    fun getAccessToken(
+        resource: String?,
+        scope: List<String>?,
+        callback: RetrieveCallback<AccessToken>,
+    ) {
+        if (!isAuthenticated()) {
+            callback.onResult(LogtoException(LogtoException.Message.NOT_AUTHENTICATED), null)
+            return
+        }
+
+        if (refreshToken == null) {
```

Comment:
check `refreshToken` only when we are about to use it? how about line 74

Reply:
updated